### PR TITLE
refactor: update cache to the new generic version

### PR DESCRIPTION
**Pull request by Stebalien**
- Adds type safety.
- Reduces allocations.
- ~Fixes the drand cache (was storing by value, but retrieving by pointer)~ (steven can't read)

### CODE REVIEW

File: chain/beacon/drand/drand.go [hunk 1]
```diff
@@ -160,16 +160,12 @@ func (db *DrandBeacon) Entry(ctx context.Context, round uint64) <-chan beacon.Re
 	return out
 }
 func (db *DrandBeacon) cacheValue(e types.BeaconEntry) {
-	db.localCache.Add(e.Round, e)
+	db.localCache.Add(e.Round, &e)
```
Comment (Stebalien) [inline chain/beacon/drand/drand.go:163]: the bug

**Comment by Stebalien** [comment]
(why now? I want to use an LRU cache in the Eth API, but want to be able to use the generic version)

File: chain/beacon/drand/drand.go (hunk 1 above)
Comment (Stebalien) [reply chain/beacon/drand/drand.go:163]: Ok, not a bug. Steven's just stupid.

**Review by arajasek — APPROVED** [review]
LGTM.

**Pull request merged**